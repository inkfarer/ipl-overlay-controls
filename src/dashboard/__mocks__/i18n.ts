import i18next from 'i18next';
import { config } from '@vue/test-utils';

jest.mock('i18next-vue', () => ({
    __esModule: true,
    useTranslation: () => ({ t: i18next.t })
}));

beforeAll(async () => {
    await i18next.init({
        lng: 'cimode',
        appendNamespaceToCIMode: true
    });
    config.global.mocks = {
        $t: i18next.t
    };
});
