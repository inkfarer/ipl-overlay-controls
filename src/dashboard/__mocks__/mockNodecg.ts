import Mock = jest.Mock;

export type ReplicantChangeHandler = (newValue?: unknown, oldValue?: unknown) => void;
export type MockReplicant = { value: unknown };
type OnFunction = (event: string, handler: ReplicantChangeHandler) => void;

export class MockNodecg {
    replicants: {[key: string]: MockReplicant};
    listeners: {[key: string]: ReplicantChangeHandler};
    sendMessage: Mock

    constructor() {
        this.replicants = {};
        this.listeners = {};
        this.sendMessage = jest.fn();
    }

    init(): void {
        const self = this;
        window.nodecg = {
            // @ts-ignore: Just a mock
            Replicant(name: string) {
                const replicantValue: { value: unknown, on: OnFunction } = {
                    value: undefined,
                    on: (event: string, handler: ReplicantChangeHandler) => {
                        self.listeners[name] = handler;
                    }
                };
                self.replicants[name] = replicantValue;
                return replicantValue;
            },
            sendMessage: self.sendMessage,
            // @ts-ignore: Just a mock
            log: {
                info: jest.fn()
            }
        };

        // @ts-ignore: It's a mock.
        window.NodeCG = {
            waitForReplicants: jest.fn().mockResolvedValue(true)
        };
    }
}
