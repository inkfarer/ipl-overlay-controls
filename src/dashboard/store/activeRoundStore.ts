import { ActiveRound, SwapColorsInternally } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { GameWinner } from 'types/enums/gameWinner';
import { SetActiveColorRequest, SetWinnerRequest, SwapRoundColorRequest } from 'types/messages/activeRound';
import { SetRoundRequest } from 'types/messages/rounds';
import { ActiveRoundGame } from 'types/activeRoundGame';

const activeRound = nodecg.Replicant<ActiveRound>('activeRound');
const swapColorsInternally = nodecg.Replicant<SwapColorsInternally>('swapColorsInternally');

export const activeRoundReps = [ activeRound, swapColorsInternally ];

export interface ActiveRoundStore {
    activeRound: ActiveRound;
    swapColorsInternally: SwapColorsInternally;
}

export const activeRoundStore = createStore<ActiveRoundStore>({
    state: {
        activeRound: null,
        swapColorsInternally: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        setWinner(store, { winner }: { winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner });
        },
        removeWinner(): void {
            nodecg.sendMessage('removeWinner');
        },
        setActiveColor(store, request: SetActiveColorRequest): void {
            nodecg.sendMessage('setActiveColor', request);
        },
        swapColors(): void {
            swapColorsInternally.value = !swapColorsInternally.value;
        },
        setWinnerForIndex(store, { index, winner }: { index: number, winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner, roundIndex: index } as SetWinnerRequest);
        },
        resetActiveRound(): void {
            nodecg.sendMessage('resetActiveRound');
        },
        setActiveRound(store, request: SetRoundRequest): void {
            nodecg.sendMessage('setActiveRound', request);
        },
        swapRoundColor(store, request: SwapRoundColorRequest): void {
            nodecg.sendMessage('swapRoundColor', request);
        },
        updateActiveGames(store, games: ActiveRoundGame[]): void {
            nodecg.sendMessage('updateActiveGames', { games });
        },
        switchToNextColor(): void {
            nodecg.sendMessage('switchToNextColor');
        },
        switchToPreviousColor(): void {
            nodecg.sendMessage('switchToPreviousColor');
        },
        async getNextAndPreviousColors(): Promise<void> {
            return nodecg.sendMessage('getNextAndPreviousColors');
        }
    }
});

export const activeRoundStoreKey: InjectionKey<Store<ActiveRoundStore>> = Symbol();

export function useActiveRoundStore(): Store<ActiveRoundStore> {
    return useStore(activeRoundStoreKey);
}
