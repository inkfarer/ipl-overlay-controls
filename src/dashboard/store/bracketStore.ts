import { Bracket } from '@tourneyview/common';
import { defineStore } from 'pinia';

const bracketData = nodecg.Replicant<Bracket | undefined>('bracketData');

export const bracketReps = [ bracketData ];

export interface BracketStore {
    bracketData: Bracket | undefined;
}

export const useBracketStore = defineStore('brackets', {
    state: () => ({
        bracketData: undefined
    } as BracketStore),
    actions: {
        resetBracketData() {
            bracketData.value = undefined;
        }
    }
});
