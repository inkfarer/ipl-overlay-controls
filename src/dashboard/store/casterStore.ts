import { NodeCGBrowser } from 'nodecg/browser';
import { Caster, Casters, RadiaSettings } from 'schemas';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

const casters = nodecg.Replicant<Casters>('casters');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');

export const castersReps = [ casters, radiaSettings ];

export interface CasterStore {
    casters: Casters;
    uncommittedCasters: Casters;
    radiaSettings: RadiaSettings;
}

const defaultCaster = {
    name: 'New Caster',
    twitter: '',
    pronouns: '?/?'
};

export const useCasterStore = defineStore('casters', {
    state: () => ({
        casters: {},
        uncommittedCasters: {},
        radiaSettings: {
            enabled: null,
            guildID: null,
            updateOnImport: null
        }
    } as CasterStore),
    actions: {
        updateCaster({ id, newValue }: { id: string, newValue: Caster }): void {
            casters.value[id] = newValue;
        },
        removeUncommittedCaster(id: string): void {
            delete this.uncommittedCasters[id];
        },
        addUncommittedCaster({ id, caster }: { id: string, caster?: Caster }): void {
            this.uncommittedCasters[id] = caster ?? defaultCaster;
        },
        addDefaultCaster(): string {
            const id = generateId();
            this.addUncommittedCaster({ id });
            return id;
        },
        async saveUncommittedCaster({ id, caster }: { id: string, caster: Caster }): Promise<string> {
            const newId = await nodecg.sendMessage('saveCaster', caster);
            this.removeUncommittedCaster(id);
            return newId;
        },
        async loadCastersFromVc(): Promise<void> {
            const result = await nodecg.sendMessage('getLiveCommentators');
            if (result.extra && result.extra.length > 0) {
                for (let i = 0; i < result.extra.length; i++) {
                    const extraCaster = result.extra[i];
                    const id = extraCaster.discord_user_id;
                    delete extraCaster.discord_user_id;
                    this.addUncommittedCaster({ id, caster: extraCaster });
                }
            }
        },
        showCasters() {
            nodecg.sendMessage('mainShowCasters');
        },
        async removeCaster(id: string): Promise<void> {
            return nodecg.sendMessage('removeCaster', { id });
        }
    }
});
