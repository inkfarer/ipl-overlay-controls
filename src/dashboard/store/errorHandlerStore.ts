import { App } from 'vue';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

export const useErrorHandlerStore = defineStore('errorHandler', {
    state: () => ({
        recentErrors: {}
    }),
    actions: {
        removeRecentError({ key }: { key: string }): void {
            delete this.recentErrors[key];
        },
        handleError({ err, info }: { err: unknown, info: string }): void {
            console.error(`Got error from '${info}': \n`, err);

            if (Object.keys(this.recentErrors).length >= 2) return;

            const id = generateId();
            this.recentErrors[id] = err;
            window.setTimeout(() => {
                delete this.recentErrors[id];
            }, 25000);
        }
    }
});

export function setUpErrorHandler(app: App<unknown>): void {
    const store = useErrorHandlerStore();
    app.config.errorHandler = (err, vm, info) => {
        store.handleError({ err, info });
    };
}
