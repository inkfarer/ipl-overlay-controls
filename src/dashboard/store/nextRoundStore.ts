import { NextRound } from 'schemas';
import { SetNextRoundRequest } from 'types/messages/rounds';
import { BeginNextMatchRequest } from 'types/messages/activeRound';
import { defineStore } from 'pinia';

const nextRound = nodecg.Replicant<NextRound>('nextRound');

export const nextRoundReps = [ nextRound ];

export interface NextRoundStore {
    nextRound: NextRound;
}

export const useNextRoundStore = defineStore('nextRound', {
    state: () => ({
        nextRound: null
    } as NextRoundStore),
    actions: {
        setShowOnStream(newValue: boolean) {
            nextRound.value.showOnStream = newValue;
        },
        beginNextMatch(request: BeginNextMatchRequest): void {
            nodecg.sendMessage('beginNextMatch', request);
        },
        setNextRound(request: SetNextRoundRequest): void {
            nodecg.sendMessage('setNextRound', request);
        }
    }
});
