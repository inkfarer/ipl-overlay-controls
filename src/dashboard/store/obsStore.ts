import { NodeCGBrowser } from 'nodecg/browser';
import { ObsCredentials, ObsData } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { SetObsDataRequest } from 'types/messages/obs';

const obsData = nodecg.Replicant<ObsData>('obsData');
const obsCredentials = nodecg.Replicant<ObsCredentials>('obsCredentials');

export const obsReps = [obsData, obsCredentials];

export interface ObsStore {
    obsData: ObsData
    obsCredentials: ObsCredentials
}

export const obsStore = createStore<ObsStore>({
    state: {
        obsData: null,
        obsCredentials: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        async connect(store, { address, password }: { address: string, password?: string }): Promise<void> {
            return nodecg.sendMessage('connectToObs', { address, password });
        },
        async setData(store, data: SetObsDataRequest): Promise<void> {
            return nodecg.sendMessage('setObsData', data);
        },
        async startGame(): Promise<void> {
            return nodecg.sendMessage('startGame');
        },
        async endGame(): Promise<void> {
            return nodecg.sendMessage('endGame');
        },
        setEnabled(store, enabled: boolean): void {
            nodecg.sendMessage('setObsSocketEnabled', enabled);
        }
    }
});

export const obsStoreKey: InjectionKey<Store<ObsStore>> = Symbol();

export function useObsStore(): Store<ObsStore> {
    return useStore(obsStoreKey);
}
