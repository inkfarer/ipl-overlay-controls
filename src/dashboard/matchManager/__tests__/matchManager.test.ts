import MatchManager from '../matchManager.vue';
import { config, mount, VueWrapper } from '@vue/test-utils';
import { ObsStatus } from 'types/enums/ObsStatus';
import { messageListeners, mockSendMessage } from '../../__mocks__/mockNodecg';
import { useCasterStore } from '../../store/casterStore';
import { createTestingPinia, TestingPinia } from '@pinia/testing';
import { useObsStore } from '../../store/obsStore';
import { GameAutomationAction } from 'types/enums/GameAutomationAction';
import { UnknownModule } from '../../../helpers/__mocks__/module';
import { IplButton, IplMessage } from '@iplsplatoon/vue-components';

describe('MatchManager', () => {
    let pinia: TestingPinia;

    config.global.stubs = {
        NextMatchStarter: true,
        IplErrorDisplay: true,
        ScoreboardEditor: true,
        IplButton: true,
        ActiveColorToggles: true,
        IplExpandingSpaceGroup: true,
        ColorEditor: true,
        ActiveMatchEditor: true,
        ScoreDisplay: true,
        SetEditor: true,
        FontAwesomeIcon: true,
        ActiveRosterDisplay: true
    };

    beforeEach(() => {
        pinia = createTestingPinia();

        config.global.plugins = [pinia];

        useObsStore().$state = {
            // @ts-ignore
            obsData: {
                status: ObsStatus.CONNECTED
            }
        };
    });

    it.each(Object.values(ObsStatus))('matches snapshot if obs status is %s', status => {
        const obsStore = useObsStore();
        obsStore.obsData.status = status;
        const wrapper = mount(MatchManager);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('handles showing casters', () => {
        const casterStore = useCasterStore();
        jest.spyOn(casterStore, 'showCasters');
        const wrapper = mount(MatchManager);

        wrapper.getComponent<typeof IplButton>('[data-test="show-casters-button"]').vm.$emit('click');

        expect(casterStore.showCasters).toHaveBeenCalled();
    });

    it('disables showing casters when message to show casters is received from nodecg and enables it after a delay', async () => {
        jest.useFakeTimers();
        const wrapper = mount(MatchManager);

        messageListeners.mainShowCasters();
        await wrapper.vm.$nextTick();

        expect(wrapper.getComponent('[data-test="show-casters-button"]').attributes().disabled).toEqual('true');

        jest.advanceTimersByTime(5000);
        await wrapper.vm.$nextTick();

        expect(wrapper.getComponent('[data-test="show-casters-button"]').attributes().disabled).toEqual('false');

        jest.useRealTimers();
    });

    it('does not show button to cancel automation action when no action is in progress', () => {
        const obsStore = useObsStore();
        obsStore.gameAutomationData = {
            actionInProgress: GameAutomationAction.NONE,
            nextTaskForAction: null
        };
        const wrapper = mount(MatchManager);

        expect(wrapper.findComponent('[data-test="cancel-automation-action-button"]').exists()).toEqual(false);
    });

    describe('when automation action is in progress', () => {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        let wrapper: VueWrapper<any>;

        beforeEach(() => {
            const obsStore = useObsStore();
            obsStore.gameAutomationData = {
                actionInProgress: GameAutomationAction.START_GAME,
                nextTaskForAction: { name: 'action!!', index: 2, executionTimeMillis: 123123 }
            };
            wrapper = mount(MatchManager);
        });

        it('handles start-stop button press', () => {
            const obsStore = useObsStore();
            obsStore.fastForwardToNextGameAutomationTask = jest.fn();

            wrapper.getComponent<typeof IplButton>('[data-test="start-stop-game-button"]').vm.$emit('click');

            expect(obsStore.fastForwardToNextGameAutomationTask).toHaveBeenCalled();
        });

        it.each(['changeScene', 'showScoreboard', 'hideScoreboard', 'showCasters', 'unknown task'])('has expected button color and label when task name is %s', async (taskName) => {
            useObsStore().gameAutomationData.nextTaskForAction.name = taskName;
            await wrapper.vm.$nextTick();
            const button = wrapper.getComponent<typeof IplButton>('[data-test="start-stop-game-button"]');

            expect((button.vm as unknown as UnknownModule).color).toEqual('blue');
            expect((button.vm as unknown as UnknownModule).label).toMatchSnapshot();
        });

        it('shows button to cancel action', () => {
            expect(wrapper.getComponent('[data-test="cancel-automation-action-button"]').isVisible()).toEqual(true);
        });

        it('handles cancel button press', () => {
            const button = wrapper.getComponent<typeof IplButton>('[data-test="cancel-automation-action-button"]');

            button.vm.$emit('click');

            expect(mockSendMessage).toHaveBeenCalledWith('cancelAutomationAction', undefined);
        });
    });

    describe('when gameplay scene is active', () => {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        let wrapper: VueWrapper<any>;

        beforeEach(() => {
            const obsStore = useObsStore();
            obsStore.gameAutomationData = {
                actionInProgress: GameAutomationAction.NONE
            };
            obsStore.obsData.currentScene = 'Gameplay Scene';
            obsStore.obsData.gameplayScene = 'Gameplay Scene';
            wrapper = mount(MatchManager);
        });

        it('handles start-stop button press', () => {
            const obsStore = useObsStore();
            obsStore.endGame = jest.fn();

            wrapper.getComponent<typeof IplButton>('[data-test="start-stop-game-button"]').vm.$emit('click');

            expect(obsStore.endGame).toHaveBeenCalled();
        });

        it('has expected button color and label', () => {
            const button = wrapper.getComponent<typeof IplButton>('[data-test="start-stop-game-button"]');

            expect((button.vm as unknown as UnknownModule).color).toEqual('red');
            expect((button.vm as unknown as UnknownModule).label).toEqual('End Game');
        });
    });

    describe('when gameplay scene is not active', () => {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        let wrapper: VueWrapper<any>;

        beforeEach(() => {
            const obsStore = useObsStore();
            obsStore.gameAutomationData = {
                actionInProgress: GameAutomationAction.NONE
            };
            obsStore.obsData.currentScene = 'another scene';
            obsStore.obsData.gameplayScene = 'Gameplay Scene';
            wrapper = mount(MatchManager);
        });

        it('handles start-stop button press', () => {
            const obsStore = useObsStore();
            obsStore.startGame = jest.fn();

            wrapper.getComponent<typeof IplButton>('[data-test="start-stop-game-button"]').vm.$emit('click');

            expect(obsStore.startGame).toHaveBeenCalled();
        });

        it('has expected button color and label', () => {
            const button = wrapper.getComponent<typeof IplButton>('[data-test="start-stop-game-button"]');

            expect((button.vm as unknown as UnknownModule).color).toEqual('green');
            expect((button.vm as unknown as UnknownModule).label).toEqual('Start Game');
        });
    });

    it.each([9001, 9500, 9999, 10000, 11000])('disables completing task when execution time is one second or less from now (now=%s)', async (now) => {
        const obsStore = useObsStore();
        obsStore.gameAutomationData = {
            actionInProgress: GameAutomationAction.START_GAME,
            nextTaskForAction: { name: 'action!!', index: 2, executionTimeMillis: 10000 }
        };
        jest.spyOn(Date.prototype, 'getTime').mockReturnValue(now);
        const wrapper = mount(MatchManager);
        const button = wrapper.getComponent<typeof IplButton>('[data-test="start-stop-game-button"]');

        expect((button.vm as unknown as UnknownModule).disabled).toEqual(true);
    });

    it('enables completing task when execution time is more than one second from now', () => {
        const obsStore = useObsStore();
        obsStore.gameAutomationData = {
            actionInProgress: GameAutomationAction.START_GAME,
            nextTaskForAction: { name: 'action!!', index: 2, executionTimeMillis: 10000 }
        };
        jest.spyOn(Date.prototype, 'getTime').mockReturnValue(8999);
        const wrapper = mount(MatchManager);
        const button = wrapper.getComponent<typeof IplButton>('[data-test="start-stop-game-button"]');

        expect((button.vm as unknown as UnknownModule).disabled).toEqual(false);
    });

    it('does not show scene configuration change warning by default', () => {
        const wrapper = mount(MatchManager);

        expect(wrapper.findComponent('[data-test="obs-scenes-changed-warning"]').exists()).toEqual(false);
    });

    it('shows warning if obs scene configuration is changed', async () => {
        const obsStore = useObsStore();
        obsStore.obsData.gameplayScene = 'Gameplay Scene';
        obsStore.obsData.intermissionScene = 'Intermission Scene';
        const wrapper = mount(MatchManager);

        messageListeners.obsSceneConfigurationChangedAfterUpdate();
        await wrapper.vm.$nextTick();

        const warning = wrapper.findComponent('[data-test="obs-scenes-changed-warning"]');
        expect(warning.exists()).toEqual(true);
        expect(warning.text()).toEqual('The OBS scene configuration has changed. Please confirm that the configured gameplay and intermission scenes (\'Gameplay Scene\' & \'Intermission Scene\') are still correct.');
    });

    it('allows to close obs scene configuration warning', async () => {
        const wrapper = mount(MatchManager);

        messageListeners.obsSceneConfigurationChangedAfterUpdate();
        await wrapper.vm.$nextTick();

        const warning = wrapper.findComponent<typeof IplMessage>('[data-test="obs-scenes-changed-warning"]');
        expect(warning.exists()).toEqual(true);

        warning.vm.$emit('close');
        await wrapper.vm.$nextTick();

        expect(wrapper.findComponent('[data-test="obs-scenes-changed-warning"]').exists()).toEqual(false);
    });
});
