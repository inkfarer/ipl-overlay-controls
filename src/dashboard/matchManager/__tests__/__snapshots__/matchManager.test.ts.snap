// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchManager matches snapshot if obs status is CONNECTED 1`] = `
"<ipl-error-display-stub class="m-b-8"></ipl-error-display-stub>
<div class="layout horizontal">
  <div class="max-width">
    <score-display-stub></score-display-stub>
    <active-color-toggles-stub class="m-t-8"></active-color-toggles-stub>
    <div data-v-11cc5f1c="" class="ipl-space color-dark m-t-8">
      <!--v-if-->
      <ipl-button-stub label="End Game" color="red" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="start-stop-game-button" class="m-b-8"></ipl-button-stub>
      <div class="layout horizontal">
        <!--v-if-->
        <ipl-button-stub label="Show casters" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="true" disableonsuccess="false" href="" data-test="show-casters-button" class="max-width"></ipl-button-stub>
      </div>
    </div>
  </div>
  <ipl-expanding-space-group-stub class="max-width m-l-8"></ipl-expanding-space-group-stub>
</div>
<set-editor-stub class="m-t-8"></set-editor-stub>"
`;

exports[`MatchManager matches snapshot if obs status is CONNECTING 1`] = `
"<ipl-error-display-stub class="m-b-8"></ipl-error-display-stub>
<div class="layout horizontal">
  <div class="max-width">
    <score-display-stub></score-display-stub>
    <active-color-toggles-stub class="m-t-8"></active-color-toggles-stub>
    <div data-v-11cc5f1c="" class="ipl-space color-dark m-t-8">
      <!--v-if-->
      <!--v-if-->
      <div class="layout horizontal">
        <!--v-if-->
        <ipl-button-stub label="Show casters" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="show-casters-button" class="max-width"></ipl-button-stub>
      </div>
    </div>
  </div>
  <ipl-expanding-space-group-stub class="max-width m-l-8"></ipl-expanding-space-group-stub>
</div>
<set-editor-stub class="m-t-8"></set-editor-stub>"
`;

exports[`MatchManager matches snapshot if obs status is NOT_CONNECTED 1`] = `
"<ipl-error-display-stub class="m-b-8"></ipl-error-display-stub>
<div class="layout horizontal">
  <div class="max-width">
    <score-display-stub></score-display-stub>
    <active-color-toggles-stub class="m-t-8"></active-color-toggles-stub>
    <div data-v-11cc5f1c="" class="ipl-space color-dark m-t-8">
      <!--v-if-->
      <!--v-if-->
      <div class="layout horizontal">
        <!--v-if-->
        <ipl-button-stub label="Show casters" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="show-casters-button" class="max-width"></ipl-button-stub>
      </div>
    </div>
  </div>
  <ipl-expanding-space-group-stub class="max-width m-l-8"></ipl-expanding-space-group-stub>
</div>
<set-editor-stub class="m-t-8"></set-editor-stub>"
`;

exports[`MatchManager when automation action is in progress has expected button color and label when task name is changeScene 1`] = `"Change Scene"`;

exports[`MatchManager when automation action is in progress has expected button color and label when task name is hideScoreboard 1`] = `"Hide Scoreboard"`;

exports[`MatchManager when automation action is in progress has expected button color and label when task name is showCasters 1`] = `"Show Casters"`;

exports[`MatchManager when automation action is in progress has expected button color and label when task name is showScoreboard 1`] = `"Show Scoreboard"`;

exports[`MatchManager when automation action is in progress has expected button color and label when task name is unknown task 1`] = `"???"`;
